client/src/main/java/org/apache/celeborn/client/read/RssInputStream.java

```diff
@@ -223,12 +223,19 @@ private void moveToNextReader() throws IOException {
     }
 
     private PartitionReader createReaderWithRetry(PartitionLocation location) throws IOException {
+      PartitionLocation lastPeer = location;
       while (fetchChunkRetryCnt < fetchChunkMaxRetry) {
         try {
           return createReader(location, fetchChunkRetryCnt, fetchChunkMaxRetry);
         } catch (Exception e) {
           fetchChunkRetryCnt++;
```

Comment:
Now every turn or no peer need sleep retryWaitMs otherwise try peer directly, if coding this way would clear than before?